packages/ra-no-code/src/ResourceConfiguration/ResourceConfigurationProvider.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import { ReactNode, useCallback, useMemo, useState } from 'react';
+import {
+    ResourceConfigurationMap,
+    ResourceConfiguration,
+    ResourceConfigurationContext,
+} from './ResourceConfigurationContext';
+
+export const ResourceConfigurationProvider = ({
+    children,
+}: {
+    children: ReactNode;
+}) => {
+    const [resources, setInternalResources] = useState<
+        ResourceConfigurationMap
+    >(() => loadConfigurationsFromLocalStorage());
+
+    const setResources = useCallback(
+        (
+            value: (
+                prevState: ResourceConfigurationMap
+            ) => ResourceConfigurationMap
+        ) => {
+            setInternalResources(prevState => {
+                const newState = value(prevState);
+
+                if (newState != undefined) { // eslint-disable-line
+                    window.localStorage.setItem(
+                        StorageKey,
+                        JSON.stringify(newState)
+                    );
+                } else {
+                    window.localStorage.removeItem(StorageKey);
+                }
+                return newState;
+            });
+        },
+        []
+    );
+
+    const addResource = useCallback(
+        (resource: ResourceConfiguration) => {
+            setResources(current => {
+                const allResources = current || {};
+                if (allResources[resource.name]) {
+                    return allResources;
+                }
+                return {
+                    ...current,
+                    [resource.name]: resource,
+                };
+            });
+        },
+        [setResources]
+    );
+
+    const updateResource = useCallback(
+        (name: string, newResource: Partial<ResourceConfiguration>) => {
+            setResources(current => {
+                const allResources = current || {};
+                const resource = allResources[name];
+                if (!resource) {
+                    return allResources;
+                }
+                const nextResources: ResourceConfigurationMap = {
+                    ...current,
+                    [name]: {
+                        ...current[name],
+                        ...newResource,
+                    },
+                };
+                return nextResources;
+            });
+        },
+        [setResources]
+    );
+
+    const removeResource = useCallback(
+        (name: string) => {
+            setResources(current => {
+                const allResources = current || {};
+                const resourceToRemove = allResources[name];
+
+                if (!resourceToRemove) {
+                    return allResources;
+                }
+                const storedData = window.localStorage.getItem(
```

Comment:
it's weird to see data manipulation code here. It's an implicit depenency on ra-data-localStorage, and this package allows for a custom storage key...

Why don't you use the dataProvider directly, by calling its deleteMany method?

Reply:
We're outside the Admin so I'd need to require the dataProvider as well but why not